server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -104,7 +104,7 @@ static void executeRank(SearchContext searchContext) throws QueryPhaseExecutionE
                         queryPhaseRankShardContext.rankWindowSize()
                     )
                 ) {
-                    QueryPhase.addCollectorsAndSearch(rankSearchContext);
+                    QueryPhase.addCollectorsAndSearch(rankSearchContext, null);
```

Comment:
I will look into this as a follow-up, need to write a test that leverages this and see if it makes sense to track timestamp from for it too, it probably does.

Reply:
Looking deeper at this, I think we are fine not tracking the time range from in this case. It's only used when multiple ranking steps are required. I don't think that we can reliably track the time range from in that case, as each sub-query may have its own independent filters. It's not that helpful either because using time range filters combined with ranking is a bit of an edge case that we don't want to focus on at the moment.